korge/src/commonMain/kotlin/com/soywiz/korge/view/animation/ParallaxDataView.kt

```diff
@@ -0,0 +1,432 @@
+
+                    // Do horizontal or vertical movement depending on parallax scrolling direction
+                    when (mode) {
+                        ParallaxMode.HORIZONTAL_PLANE -> {
+                            addUpdater {
+                                // Calculate the offset for the inner scrolling of the layer
+                                x += (deltaX * speedX + selfSpeedX) * it.milliseconds
+                            }
+                        }
+                        ParallaxMode.VERTICAL_PLANE -> {
+                            addUpdater {
+                                // Calculate the offset for the inner scrolling of the layer
+                                y += (deltaY * speedY + selfSpeedY) * it.milliseconds
+                            }
+                        }
+                        ParallaxMode.NO_PLANE -> {
+                            addUpdater {
+                                // Calculate the offset for the inner scrolling of the layer
+                                x += (deltaX * speedX + selfSpeedX) * it.milliseconds
+                                y += (deltaY * speedY + selfSpeedY) * it.milliseconds
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    init {
+        // Only the base container for all view objects needs to be scaled
+        this.scale = scale
+
+        // First create background layers in the back
+        constructLayer(data.backgroundLayers, data.config.backgroundLayers, data.config.mode, smoothing)
+
+        // Then construct the two parallax planes with their attached layers
+        if (data.config.mode != ParallaxMode.NO_PLANE) {
+            constructParallaxPlane(
+                data.parallaxPlane,
+                data.attachedLayers,
+                data.config.parallaxPlane,
+                data.config.mode == ParallaxMode.HORIZONTAL_PLANE,
+                smoothing
+            )
+            // Do horizontal or vertical movement depending on parallax scrolling direction
+            if (data.config.mode == ParallaxMode.HORIZONTAL_PLANE) {
+                // Move parallax plane inside borders
+                addUpdater {
+                    // Sanity check of diagonal movement - it has to be between 0.0 and 1.0
+                    diagonal = diagonal.clamp(0.0, 1.0)
+                    y = -(diagonal * (parallaxLayerSize - data.config.size.height))
+                }
+            } else {
+                addUpdater {
+                    diagonal = diagonal.clamp(0.0, 1.0)
+                    x = -(diagonal * (parallaxLayerSize - data.config.size.width))
+                }
+            }
+        }
+
+        // Last construct the foreground layers on top
+        constructLayer(data.foregroundLayers, data.config.foregroundLayers, data.config.mode, smoothing)
+    }
+}
+
+suspend fun VfsFile.readParallaxDataContainerFromAseFile(
```

Comment:
Could it be renamed to `readParallaxDataContainer` and pass the file format as parameter? That way it could be used from example from a `.psd` or a `.kra` file or a custom format provided by someone using other program

Reply:
Yes, that's also fine.

Reply:
Done :)